onnxruntime/core/platform/windows/env.cc

```diff
@@ -124,52 +147,171 @@ class WindowsEnv : public Env {
   }
 
   int GetNumCpuCores() const override {
-    SYSTEM_LOGICAL_PROCESSOR_INFORMATION buffer[256];
-    DWORD returnLength = sizeof(buffer);
-    if (GetLogicalProcessorInformation(buffer, &returnLength) == FALSE) {
-      // try GetSystemInfo
-      SYSTEM_INFO sysInfo;
-      GetSystemInfo(&sysInfo);
-      if (sysInfo.dwNumberOfProcessors <= 0) {
-        ORT_THROW("Fatal error: 0 count processors from GetSystemInfo");
+    int num_cores = 0;
+    for (const auto& group_info : group_info_vec_) {
+      num_cores += group_info.ActiveProcessorCount;
+    }
+    return num_cores;
+  }
+
+  size_t GetDefaultThreadpoolSetting(std::vector<uint64_t>& affinities) const override {
+    ORT_ENFORCE(sizeof(KAFFINITY) <= sizeof(uint64_t), "KAFFINITY is bigger than uint64_t, cannot save it to an uint64_t vector");
+    affinities.clear();
+    KAFFINITY group_id = 0;
+    for (const auto& group_info : group_info_vec_) {
+      size_t bit_offset = 0;
+      constexpr size_t bit_limit = sizeof(KAFFINITY) << 3;
```

Comment:
>sizeof(KAFFINITY) << 3; [](http://example.com/codeflow?start=35&length=23)

1)This is not a perf critical code. Bitwise operations are harder to read.
2) So 8 divided by 8 would always be 1?

Reply:
let me try CHAT_BIT then, thanks!